app/views/handlers/list_collector.py

```diff
@@ -21,6 +23,18 @@ def get_next_location_url(self):
             )
             return add_url
 
+        if not self._is_list_collector_complete() and (incomplete_block := next(self._get_incomplete_repeating_block_ids(), None)):
```

Comment:
We don't need `_is_list_collector_complete`, if `_get_incomplete_repeating_block_ids` returns an id then we know it is not complete. Then it is consistent with how we currently do it.

So logic should just be `if incomplete_block: return url`

Reply:
Yep good catch! Have removed.